packages/next/build/webpack/config/blocks/css/overrideCssConfiguration.ts

```diff
@@ -1,20 +1,29 @@
 import { webpack } from 'next/dist/compiled/webpack/webpack'
+import type { AcceptedPlugin } from 'postcss'
 import { getPostCssPlugins } from './plugins'
 
-export async function __overrideCssConfiguration(
+export function __overrideCssConfiguration(
   rootDirectory: string,
   isProduction: boolean,
   config: webpack.Configuration
 ) {
-  const postCssPlugins = await getPostCssPlugins(rootDirectory, isProduction)
+  let postCssPlugins: AcceptedPlugin[] | undefined
 
   function patch(rule: webpack.RuleSetRule) {
     if (
       rule.options &&
       typeof rule.options === 'object' &&
-      typeof rule.options.postcssOptions === 'object'
+      (typeof rule.options.postcssOptions === 'object' ||
+        typeof rule.options.postcssOptions === 'function')
```

Comment:
Was it intentional to add new behavior here? I don't where this was added for in regards to performance.

Reply:
postcss-loader allows to pass a Function as `postcssOptions`. I allows to patch that too.

In some other places I use this to load postcss options only when they are actually used.

But something is broken with that as some postcss related tests break, e. g. `Bad CSS Customization`

Reply:
Yea, I believe we specifically have tests that ensure a function isn't allowed here. I'd revert this change for now.